pypesto/sample/pymc3.py

```diff
@@ -30,98 +32,135 @@ class Pymc3Sampler(Sampler):
     step_function:
         A pymc3 step function, e.g. NUTS, Slice. If not specified, pymc3
         determines one automatically (preferable).
+    cache_gradients:
+        If `True`, evaluate gradient together with objective value.
+        This results in a modest speed-up for `pymc3` samplers using gradient
+        information (such as NUTS).
+        It can be safely kept on even when using other samplers.
+    vectorize:
+        If `True`, all free parameters will be gathered in a single vector
+        inside the `pymc3.Model` object. This results in a small speed-up,
+        especially if the objective computation time is small or if simpler
+        step functions (such as `pymc3.Metropolis`) are used.
+    remap_to_reals:
+        If `True`, parameters will be remapped to the whole real line for
+        use in PyMC3 (this is required by the NUTS sampler). If not specified,
+        it will be set to `True` if NUTS is used and to `False` otherwise.
     **kwargs:
         Options are directly passed on to `pymc3.sample`.
     """
 
-    def __init__(self, step_function=None, **kwargs):
+    def __init__(self, cache_gradients: bool = True,
+                       vectorize: bool = True,
+                       remap_to_reals: Optional[bool] = None, **kwargs):
+
+        if pm is None:
+            raise Exception('Please install the pymc3 package ' \
+                            'in order to use the Pymc3Sampler sampler.')
+
         super().__init__(kwargs)
```

Comment:
```suggestion
        super().__init__(**kwargs)
```

Reply:
This looks like a mistake but it is actually correct, because the `Sampler` base class receives options as a dictionary. It may be best to pass options for the derived classes in the same way, especially since all other samplers seem to use the dictionary approach. @yannikschaelte's original implementation of the PyMC3 sampler used keyword arguments and I have opted not to change it for the moment.

Reply:
It would also be fine to change to **kwargs for all, I think often that is nicer to use.

Reply:
I personally also like the `**kwargs` solution better. We should maybe bring it up and discuss with whom is responsible for the other samplers.

Reply:
That would be @elbaraim @dilpath @erikadudki + me. Fine for me.

Reply:
fine for me, as well.

Reply:
same :+1:

Reply:
Ah didn't realise it was different here. Either option is fine for me